ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -686,9 +687,25 @@ private static ImportTableDesc getBaseCreateTableDescFromTable(String dbName,
         loadTableWork.setInheritTableSpecs(false);
         moveWork.setLoadTableWork(loadTableWork);
       }
+      Task<?> loadPartTask = TaskFactory.get(moveWork, x.getConf());
+      if (performOnlyMove) {
+        if (addPartTask != null) {
+          addPartTask.addDependentTask(loadPartTask);
+        }
+        x.getTasks().add(loadPartTask);
+        return addPartTask == null ? loadPartTask : addPartTask;
```

Comment:
can be simplified. There is already a not null check above

Reply:
Yes, but then we will have to have two return statements which I wanted to avoid.

Reply:
There are already multiple return statements in the same method.